distributor/lru.go

```diff
@@ -32,8 +32,12 @@ func (lru *cache) Put(key string, value interface{}) {
 	}
 	lru.mut.Lock()
 	defer lru.mut.Unlock()
-	if _, ok := lru.get(key); ok {
-		return
+	if v, ok := lru.get(key); ok {
+		if v == value {
+			return
+		} else {
+			lru.priority.Remove(lru.priority.Front())
```

Comment:
Yes, I admit that it has a side effect as you mentioned. But I think golang's [container/list](https://golang.org/pkg/container/list/) doesn't have the "efficient" find function. Do you have any idea to find the element to remove efficiently?